pkg/manifests/config.go

```diff
@@ -66,6 +66,19 @@ func (c Config) IsStorageConfigured() bool {
 	return prometheusK8sConfig.VolumeClaimTemplate != nil
 }
 
+func (c Config) InconsistentAlermanagerConfigurations() bool {
+	if c.ClusterMonitoringConfiguration != nil && c.UserWorkloadConfiguration != nil {
+		alertmanagerMainConfig := c.ClusterMonitoringConfiguration.AlertmanagerMainConfig
+		alertmanagerUWMConfig := c.UserWorkloadConfiguration.Alertmanager
+		if alertmanagerMainConfig != nil && alertmanagerUWMConfig != nil {
+			if alertmanagerMainConfig.EnableUserAlertManagerConfig && alertmanagerUWMConfig.Enabled {
+				return true
+			}
+		}
+	}
```

Comment:
We can avoid nesting by exiting early. e.g. 

```go

if c.ClusterMonitoringConfiguration == nil || c.UserWorkloadConfiguration == nil {
  return false
}

mainConfig := c.ClusterMonitoringConfiguration.AlertmanagerMainConfig
uwmConfig := c.UserWorkloadConfiguration.Alertmanager

if mainConfig == nil || uwmConfig == nil {
  return false
}

return mainConfig.EnableUserAlertManagerConfig && uwmConfig.Enabled

Reply:
This looks wayy better! Thank you!